shell/platform/darwin/macos/framework/Source/FlutterResizeSynchronizer.mm

```diff
@@ -67,10 +67,18 @@ - (void)beginResize:(CGSize)size notify:(dispatch_block_t)notify {
 
   _waiting = YES;
 
-  _condBlockRequestCommit.wait(lock, [&] { return _pendingCommit; });
+  const int wait_time_millis = 500;
+  bool request_commit_success = _condBlockRequestCommit.wait_for(
+      lock, std::chrono::milliseconds(wait_time_millis), [&] { return _pendingCommit; });
+  if (!request_commit_success) {
+    NSLog(@"Commit wasn't requested after %d ms, no frame was requested."
+          @" Releasing the platform thread.",
+          wait_time_millis);
+  } else {
+    [_delegate resizeSynchronizerFlush:self];
+    [_delegate resizeSynchronizerCommit:self];
+  }
 
-  [_delegate resizeSynchronizerFlush:self];
-  [_delegate resizeSynchronizerCommit:self];
   _pendingCommit = NO;
```

Comment:
If the frame comes in slightly late, do we risk screwing up the subsequent resize's wait? I'm finding it hard to reason about the potential ripple effects of ignoring the fact that we didn't get the release we were expecting, so this needs *thorough* comments explaining why it's safe.